File: java/src/test/java/mrk/hackerrank/HackerRank01Test.java
package mrk.hackerrank;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

public class HackerRank01Test {

    private static Stream<Arguments> birthdayCakeCandlesArguments() {
        return Stream.of(
                Arguments.of("max appears once", new int[]{1, 2, 3, 4}, 1),
                Arguments.of("all elements equal", new int[]{1, 1, 1, 1}, 4),
                Arguments.of("max appears multiple times", new int[]{3, 1, 3, 2}, 2),
                Arguments.of("empty array", new int[]{}, 0)
        );
    }

    @ParameterizedTest(name = "{0}")
    @MethodSource("birthdayCakeCandlesArguments")
    public void birthdayCakeCandles(String name, int[] input, int output) {
        Assertions.assertEquals(output, new HackerRank01().birthdayCakeCandles(input));
    }
}

File: kotlin/src/test/kotlin/HackerRank01Test.kt
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource

class HackerRank01Test {
    @ParameterizedTest(name = "{0}")
    @MethodSource("hackerRank01Data")
    fun test_hackerRank01(
        name: String,
        input: Array<Int>,
        output: Int,
    ) {
        assertEquals(output, hackerRank01(input))
    }

    companion object {
        @JvmStatic
        fun hackerRank01Data() =
            listOf(
                Arguments.of("max appears once", arrayOf(1, 2, 3, 4), 1),
                Arguments.of("all elements equal", arrayOf(1, 1, 1, 1), 4),
                Arguments.of("max appears multiple times", arrayOf(3, 1, 3, 2), 2),
                Arguments.of("empty array", emptyArray<Int>(), 0),
            )
    }
}
